### PR TITLE
Prefill return first token

### PR DESCRIPTION
**Pull request by jwyang-google**
Modify Jetstream to make prefill return first token. Need to check in together with https://github.com/google/JetStream/pull/105

### CODE REVIEW

**Review by JoeZijunZhou — APPROVED** [review]
LGTM!

File: MaxText/maxengine.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ class DecodeState:
   generate_cache: jax.Array
   generate_cache_index: int
   generate_lengths: jax.Array
+  generated_token: jax.Array
```
Comment (JoeZijunZhou) [inline MaxText/maxengine.py:52]: is it `generated_tokens`? BTW seems the fields of DecodeState doesn't match in the code?

Comment (jwyang-google) [reply MaxText/maxengine.py:52]: actually in the code, the "generated_tokens" are generated_lengths here, "tokens" are corresponding to generated_token here.

Comment (JoeZijunZhou) [reply MaxText/maxengine.py:52]: Should the field name match to each other? Otherwise, it's not using the pre-defined fields?

**Review by gobbleturk — APPROVED** [review]

**Pull request merged**